app/acknowledgement/process-invalid.js

```diff
@@ -1,26 +1,16 @@
 const db = require('../data')
-const { addHold } = require('../holds')
-const createSchedule = require('../inbound/create-schedule')
-const getExistingHold = require('./get-existing-hold.js')
-const getExistingSchedule = require('./get-existing-schedule')
-const getHoldCategoryId = require('./get-hold-category-id')
+const { getHoldCategoryId } = require('../holds')
 const getHoldCategoryName = require('./get-hold-category-name')
 const invalidatePaymentRequests = require('./invalidate-payment-requests')
+const holdAndReschedule = require('../reschedule')
 
 const processInvalid = async (schemeId, paymentRequestId, frn, message) => {
   const transaction = await db.sequelize.transaction()
   try {
     await invalidatePaymentRequests(paymentRequestId, transaction)
     const holdCategoryName = getHoldCategoryName(message)
     const holdCategoryId = await getHoldCategoryId(schemeId, holdCategoryName, transaction)
-    const existingHold = await getExistingHold(holdCategoryId, frn)
-    if (!existingHold) {
-      await addHold(frn, holdCategoryId)
-    }
-    const existingSchedule = await getExistingSchedule(paymentRequestId, transaction)
-    if (!existingSchedule) {
-      await createSchedule(schemeId, paymentRequestId)
-    }
+    await holdAndReschedule(schemeId, paymentRequestId, holdCategoryId, frn, transaction)
```

Comment:
The steps originally here to apply hold and reschedule requests following a BACS Rejection are identical to those needed for this change.  Therefore moving to a common node module for both sets of capability to use.

Reply:
Good job on passing transaction through to addHold